tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Controllers/AccountControllerTests.cs

```diff
@@ -0,0 +1,106 @@
+                .Create()
+                .Build();
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.GetLogoutRequestAsync(It.IsAny<string>()))
+                .ReturnsAsync(expectedLogoutRequest);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            const string expectedLogoutId = "123";
+            var actual = await sut.Logout(expectedLogoutId);
+            
+            actual.Should().BeEquivalentTo(new RedirectResult(expectedLogoutRequest.PostLogoutRedirectUri));
+        }
+
+        [Test]
+        public async Task Logout_WhenLogoutIdIsNotNullOrEmpty_LogoutServiceSignOutCalledOnce()
+        {
+            var expectedLogoutRequest = LogoutRequestBuilder
+                .Create()
+                .Build();
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.SignOutAsync(It.IsAny<LogoutRequest>()))
+                .Returns(Task.CompletedTask);
+            logoutServiceMock.Setup(x => x.GetLogoutRequestAsync(It.IsAny<string>()))
+                .ReturnsAsync(expectedLogoutRequest);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            const string expectedLogoutId = "123";
+            await sut.Logout(expectedLogoutId);
+
+            logoutServiceMock.Verify(x => x.SignOutAsync(It.Is<LogoutRequest>(actual => actual.Equals(expectedLogoutRequest))), Times.Once);
+        }
+
+        [Test]
+        public async Task Logout_WhenLogoutIdIsNotNullOrEmpty_LogoutServiceGetPostLogoutRedirectUriCalledOnce()
+        {
+            var expectedLogoutRequest = LogoutRequestBuilder
+                .Create()
+                .Build();
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.GetLogoutRequestAsync(It.IsAny<string>()))
+                .ReturnsAsync(expectedLogoutRequest);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            const string expectedLogoutId = "123";
+            await sut.Logout(expectedLogoutId);
+
+            logoutServiceMock.Verify(x => x.GetLogoutRequestAsync(
+                It.Is<string>(actual => actual == null)), Times.Once);
+        }
+
+        [Test]
+        public void Logout_WhenLogoutIdIsNull_ShouldThrowArgumentNullException()
+        {
+            Assert.ThrowsAsync<ArgumentNullException>(async () =>
+            {
+                using var sut = new AccountControllerBuilder()
+                    .Build();
+
+                await sut.Logout(null);
+            });
+        }
+
+        [Test]
+        public void Logout_WhenLogoutIdIsEmpty_ShouldThrowArgumentNullException()
```

Comment:
Could this test and `Logout_WhenLogoutIdIsNull_ShouldThrowArgumentNullException` be combined into single test using test cases? If so, is it worth adding a test case for white space too?